services/robots.service.js

```diff
@@ -0,0 +1,22 @@
+const request = require('request');
+
+const requestRobotsTxtFiles = (robotTxtFileUrls) => {
+    return new Promise((resolve, reject) => {
+      request(robotTxtFileUrls, (err, res, body) => {
+        if (err) reject(err);  
+        const space = /\s+/;
+        const siteMapGzippedUrlFormat = /^https:.*(xml.gz)$/;
+        const siteMapUrlFormat = /^https:.*(xml)$/;
+        const bodyArr = body.split(space);
+        const urlArr = [];
+        for (const body of bodyArr) { 
+          if(siteMapUrlFormat.test(body) || siteMapGzippedUrlFormat.test(body)) {
+            urlArr.push(body);
+          };
+        };
+        resolve(urlArr);
+      });
```

Comment:
can be simplified as filter function:
```
const urlArr = body.split(/\s+/).filter(body => /^https:.*(xml)$/.test(body) || /^https:.*(xml.gz)$/.test(body));

resolve(urlArr);
```
- I would declare the regex outside of the function as well :)